tests/integration/s2n_handshake_test_s_client.py

```diff
@@ -285,6 +304,10 @@ def try_handshake(endpoint, port, cipher, ssl_version, server_cert=None, server_
         if find_expected_cipher(expected_cipher, s_client_out) != 0:
             return -1
 
+    if strict_hostname is True:
+        if validate_hostname(s_client_out) != 0:
+            return -1
+
```

Comment:
Each test case should know the *specific* certificate that the server should serve. I'll try to think of a way to do a byte-for-byte validation of the expected server cert sent.